tsconfig.json

```diff
@@ -11,6 +11,7 @@
     "allowSyntheticDefaultImports": true,
     "noEmit": true,
     "resolveJsonModule": true,
+    "skipLibCheck": true,
```

Comment:
This was needed to avoid the 46 errors `tsc` spits out from `node_moduels/@apache-rrow/es5-cjs`

Reply:
@trxcllnt Do you have any input on this - why are we getting typescript errors requiring us to skip lib checks?

Reply:
These should be resolved as part of https://github.com/apache/arrow/pull/10371 in  5.0.0. TS 4.2+ broke some more stuff blocking us from updating, but I'm removing the broken things in that PR.